### PR TITLE
add list of supported api's for baremetal

### PR DESCRIPTION
**Pull request by aashishc1988**
Adding list of api's that are supported on baremetal. The items in the list do not link to any webpage as of now.

### CODE REVIEW

File: docs/baremetal_api.md [hunk 1]
```diff
@@ -1 +1,43 @@
...
+<h3 id="digital-interfaces">Digital Interfaces</h3>
+- Serial - Serial/UART bus.
+
+- SPI - SPI bus master.
+- SPISlave - SPI bus slave.
+
+- I2C - I²C bus master.
+- I2CSlave - I²C bus slave.
+
+- CAN - Controller-area network bus.
```
Comment (SenRamakri) [inline docs/baremetal_api.md:47]: @aashishc1988  - Was CAN supported with MbedOS2?

**Comment by AnotherButler** [comment]
Does this one also require [ARMmbed/mbed-os#9800](https://github.com/ARMmbed/mbed-os/pull/9800)?

File: docs/baremetal_api.md (hunk 1 above)
Comment (SenRamakri) [reply docs/baremetal_api.md:47]: @aashishc1988 confirmed that CAN works, so we are ok.

**Comment by SenRamakri** [comment]
@AnotherButler @aashishc1988 - When I viewed this file in rendered form, it appears to be missing the bullet for the first item after a heading, is this expected?

@aashishc1988 - I see that the API listing is missing DigitalOut.h? Can we add that as well? Or is it not supported for some reason?

**Comment by aashishc1988** [comment]
> @aashishc1988 - I see that the API listing is missing DigitalOut.h? Can we add that as well? Or is it not supported for some reason?

We are listing DigitalOut.

**Comment by AnotherButler** [comment]
Good catch, @SenRamakri The formatting is now updated.

**Review by SenRamakri — APPROVED** [review]

**Comment by bulislaw** [comment]
Where do you want to put it in handbook exactly? I would merge it with the explanation i requested in #964

**Review by 0xc0170 — APPROVED** [review]

**Review by AnotherButler — APPROVED** [review]

**Comment by AnotherButler** [comment]
We've created a book for this for now. The introduction and comparison will go at the top of this page: https://os.mbed.com/docs/mbed-os/development/bare-metal/index.html

**Pull request merged**